llama-index-core/llama_index/core/memory/chat_summary_memory_buffer.py

```diff
@@ -0,0 +1,245 @@
+from typing import Any, Callable, Dict, List, Optional
+
+from pydantic.fields import PrivateAttr
+
+from llama_index.core.base.llms.types import ChatMessage, MessageRole
+from llama_index.core.bridge.pydantic import Field, root_validator
+from llama_index.core.llms.llm import LLM
+from llama_index.core.memory.types import DEFAULT_CHAT_STORE_KEY, BaseMemory
+from llama_index.core.storage.chat_store import BaseChatStore, SimpleChatStore
+from llama_index.core.utils import get_tokenizer
+
+DEFAULT_TOKEN_LIMIT_FULL_TEXT_RATIO = 0.75
+DEFAULT_TOKEN_LIMIT_FULL_TEXT = 2000
+SUMMARIZE_PROMPT = "The following is a conversation between the user and assistant. Write a concise summary about the contents of this conversation."
+
+
+# TODO: Add option for last N user/assistant history interactions instead of token limit
+class ChatSummaryMemoryBuffer(BaseMemory):
+    """Buffer for storing chat history that uses the full text for the latest
+    {token_limit_full_text}.
+
+    All older messages are iteratively summarized using the {summarizer_llm} provided, with
+    the max number of tokens defined by the {summarizer_llm}.
+
+    User can specify whether initial tokens (usually a system prompt)
+    should be counted as part of the {token_limit_full_text}
+    using the parameter {count_initial_tokens}.
+
+    This buffer is useful to retain the most important information from a
+    long chat history, while limiting the token count and latency
+    of each request to the LLM.
+    """
+
+    token_limit_full_text: int
+    count_initial_tokens: bool = False
+    summarizer_llm: Optional[LLM] = None

```

Comment:
for consistency with the rest of the framework, lets call this field just `llm`

Reply:
I agree. Updated.